static/src/javascripts/projects/common/modules/discussion/comment-count.spec.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import fetchJson from 'lib/fetch-json';
+import commentCount from './comment-count';
+
+jest.mock('lib/fastdom-promise');
+
+jest.mock('lib/fetch-json', () =>
+    jest.fn(() =>
+        Promise.resolve({
+            counts: [
+                { id: '/p/3ghkT', count: 0 },
+                { id: '/p/3ghv5', count: 100 },
+                { id: '/p/3ghx3', count: 200 },
+                { id: '/p/3gh4n', count: 300 },
+                { id: '/p/3ghNp', count: 400 },
+            ],
+        })
+    )
+);
+
+const fetchJsonSpy: any = fetchJson;
+
+// Not yet implemented, see https://github.com/tmpvar/jsdom/issues/961
+HTMLElement.prototype.dataset = {};
+
+describe('Comment Count', () => {
+    beforeEach(() => {
+        if (document.body) {
+            document.body.innerHTML = `<div class="comment-trails">
+                <div class="trail" data-discussion-closed="true" data-discussion-id="/p/3ghkT"><a href="/article/1">1</a></div>
+                <div class="trail" data-discussion-id="/p/3ghv5"><a href="/article/1">1</a></div>
+                <div class="trail" data-discussion-id="/p/3ghx3"><a href="/article/2">1</a></div>
+                <div class="trail" data-discussion-id="/p/3gh4n"><a href="/article/3">1</a></div>
+                <div class="trail" data-discussion-id="/p/3ghv5"><a href="/article/4">4</a></div>
+                <div class="trail" data-commentcount-format="content" data-discussion-id="/p/3ghNp"><a href="/article/3">1</a></div>
+                <div class="trail" data-commentcount-format="contentImmersive" data-discussion-id="/p/3ghNp"><a href="/article/4">4</a></div>
+            </div>`;
+        }
+
+        // Workaround to support dataset lookups
+        const trails = document.getElementsByClassName('trail');
+
+        [...trails].forEach(trail => {
+            trail.dataset = {};
+
+            const data = {
+                discussionClosed: trail.getAttribute('data-discussion-closed'),
+                discussionId: trail.getAttribute('data-discussion-id'),
+                commentcountFormat: trail.getAttribute(
+                    'data-commentcount-format'
+                ),
+            };
+
+            Object.keys(data).forEach(key => {
+                if (data[key]) {
+                    trail.dataset[key] = data[key];
+                }
+            });
```

Comment:
Why not set these properties in `trail.dataset` directly?

Reply:
To appease flow. I did as you suggested initially but flow complained as the type for dataset is `dataset: {[key:string]: string}` and the result of the `getAttribute` calls could be null. I did this so only `getAttribute` calls calls that returned a value are added to the dataset.

Reply:
Ah, I see, that makes sense. Thanks for clarifying 😄